store.go

```diff
@@ -2808,7 +2833,8 @@ func (s *store) Image(id string) (*Image, error) {
 	if err != nil {
 		return nil, err
 	}
-	for _, store := range append([]ROImageStore{istore}, istores...) {
+	for _, s := range append([]ROImageStore{istore}, istores...) {
```

Comment:
Why make this change?

Reply:
`&s` always point to the same memory as the loop variable is allocated only once, even for non-primitive types. What happens when using `defer s.Unlock()` in the loop is that we call unlock the very same lock (always the last item in the slice) for `len(append([]ROImageStore{istore}, istores...))` times.

Ultimately, I want to remove all `defer lock.Unlock()` occurances in the code as it delays releasing the lock. The "trick" used in this commit is allocate a new pointer on the heap `store := s` and use that in the defer statement.

Note that the issue of using loop variables in go applies to other cases as well. I wrote a blog about that last April (https://vrothberg.github.io/2018/04/02/pointers-in-golang-considered-harmful.html).

Reply:
Note that this particular problem of unlocking the same lock multiple times predates this PR. I am a bit surprised that it did not cause (major) issues yet as it implies that some lock remained unlocked and others have been unlocked too often.